### ISSUE TITLE
flaky test: `[sig-storage] virtiofs VirtIO-FS with multiple PVCs [Serial] should be successfully started and accessible (unprivileged virtiofsd) `

### ISSUE THREAD

**Issue by dhiller** [kubevirt/kubevirt#13141]
# What happened

Flaky test detected: `[sig-storage] virtiofs VirtIO-FS with multiple PVCs [Serial] should be successfully started and accessible (unprivileged virtiofsd) ` [1] [2]

/kind flake
/priority critical-urgent

/sig compute

FYI @xpivarc asking you whether the storage-root test lane is still relevant?

[1]: https://storage.googleapis.com/kubevirt-prow/reports/flakefinder/kubevirt/kubevirt/flakefinder-2024-10-29-024h.html#row4
[2]: https://storage.googleapis.com/kubevirt-prow/reports/flakefinder/kubevirt/kubevirt/flakefinder-2024-10-28-024h.html#row5

## Additional context

# Flake Action Plan

**As the assignee**, **thoroughly review the issue** and put the resulting report as comment into this issue.

### **The flake is a bug**

* Add label commenting `/triage accepted`
* **Create a PR to fix the bug**
* Reference this issue in the PR
* Keep this issue open until the testcase does not fail anymore

### **The flake is non-critical or an issue that is hard to fix**
* **Quarantine the test** by creating a pull request assigning [QUARANTINE] tag to test name and [Quarantine decorator](https://github.com/kubevirt/kubevirt/blob/f85a8117b8a90fd913ec0719faae9506866d1525/tests/decorators/decorators.go#L7) to the test
* Reference this issue on the PR

### There was an infrastructure issue

An infra issue is anything "below" the testcase

* Add label commenting `/triage infra-issue`
* Close the issue, adding a comment with details about the infrastructure issue.

After the flake has been fixed, document the learning from it inside the fix PR

**Comment by dhiller** [comment]
@akalenyu do you happen to know whether someone is looking at this issue?

**Comment by akalenyu** [comment]
hey, maybe we could use some resilience in https://github.com/kubevirt/kubevirt/blob/cfffc52a7f9aa9ec9af677e57a5faa8419f41b15/tests/virtiofs/datavolume.go#L131-L136 but I also see that etcd was restarting during the failures.

Let's see if this keeps failing after https://github.com/kubevirt/project-infra/pull/3717 
(I couldn't find a failure from last 14 days)

**Comment by dhiller** [comment]
@akalenyu thanks for chiming in - feel free to close this one if it seems fixed.

**Comment by dhiller** [comment]
@dhiller to take a look whether this one still fails

**Comment by dhiller** [comment]
@akalenyu please note this one is failing on the sig-storage-root lane: https://testgrid.k8s.io/kubevirt-periodics#periodic-kubevirt-e2e-k8s-1.29-sig-storage-root

**Comment by dhiller** [comment]
@akalenyu do we want to remove the sig-storage-root lanes from our reports? IIUC there’s little to no interest in general?

**Comment by akalenyu** [comment]
> @akalenyu do we want to remove the sig-storage-root lanes from our reports? IIUC there’s little to no interest in general?

Yeah, I don't think we're commited to maintaining rootful kubevirt at this point

**Comment by dhiller** [comment]
@rthallisey since you are one of the interested parties in the root lane, would someone from your side have capacity to look at this?

**Comment by dhiller** [comment]
@jcanocan or @germag would you be interested in looking at the issue?

**Comment by dhiller** [comment]
@alicefr I figure https://github.com/kubevirt/kubevirt/pull/13442 will be fixing this issue, right?

**Comment by alicefr** [comment]
Is it really a flaky test? If we don't enable the "Root" feature gate, then it should not succeed at all

**Comment by germag** [comment]
> Is it really a flaky test? If we don't enable the "Root" feature gate, then it should not succeed at all

https://github.com/kubevirt/kubevirt/pull/11997 removed the privileged virtiofsd, so now the `Root` FG is not required any more

**Comment by jcanocan** [comment]
I've been taking a look at this test. Looks like the problem may be here: 

https://github.com/kubevirt/kubevirt/blob/fbe93fcc87a249a8d94c553aecf8e13c810252d3/tests/virtiofs/datavolume.go#L96-L108

In the logs, we can find the following error:

```
[   14.126444] SELinux: (dev virtiofs, type virtiofs) has no security xattr handler
[   14.130986] SELinux: (dev virtiofs, type virtiofs) falling back to genfs
[   13.338244] cloud-init[768]: touch: cannot touch '/mnt/virtiofs_pvc-194ca23/virtiofs_test': Permission denied
[   14.147984] SELinux: (dev virtiofs, type virtiofs) has no security xattr handler
[   14.152604] SELinux: (dev virtiofs, type virtiofs) falling back to genfs
[   13.358422] cloud-init[768]: touch: cannot touch '/mnt/virtiofs_pvc-241cd36/virtiofs_test': Permission denied
```

https://storage.googleapis.com/kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.31-sig-storage-root/1881887223956115456/artifacts/k8s-reporter/1/pods/1_kubevirt-test-default1_virt-launcher-testvmi-f27gv-7xxn9-guest-console-log.log

I will keep you posted.

**Comment by dhiller** [comment]
@jcanocan any updates on your investigation?

**Comment by jcanocan** [comment]
> [@jcanocan](https://github.com/jcanocan) any updates on your investigation?

No, I don't have any updates yet. I didn't have the time to further investigate this.

**Comment by jcanocan** [comment]
I've taken a close look the problem is, in this lane, VMIs are executed with RuntimeUser = 0 (consequence of `Root` FG). This leads PVC Filesystems to be owned by user id 0:

```
[root@vm-pvc-virtiofs fedora]# mount -t virtiofs pvc-virtiofs /mnt/pvc/
[   79.827551] SELinux: (dev virtiofs, type virtiofs) has no security xattr handler
[   79.830972] SELinux: (dev virtiofs, type virtiofs) falling back to genfs
[root@vm-pvc-virtiofs fedora]# cd /mnt/
[root@vm-pvc-virtiofs mnt]# ls -lah
total 4.0K
drwxr-xr-x. 1 root root    6 Feb 10 09:57 .
dr-xr-xr-x. 1 root root  138 May  4  2022 ..
drwxr-xr-x. 2 root root 4.0K Feb 10 09:53 pvc
[root@vm-pvc-virtiofs mnt]# ls -lah pvc/
total 4.0K
drwxr-xr-x. 2 root root 4.0K Feb 10 09:53 .
drwxr-xr-x. 1 root root    6 Feb 10 09:57 ..
```

Currently, PVCs shared with virtiofs can only be written inside the VM if shared with user id 107 or have enough permissions. This is a known and documented limitation: https://kubevirt.io/user-guide/storage/disks_and_volumes/#sharing-directories-with-vms 
This leads the VM to be unable to write the `virtiofs_test` test file stated in previous comment https://github.com/kubevirt/kubevirt/issues/13141#issuecomment-2607566025.

We are working on a fix.

**Comment by dhiller** [comment]
@jcanocan any updates on this matter?

**Comment by jcanocan** [comment]
Hi @dhiller!
All changes merged
Looks like the test is working again! 🎉
